example/lib/main.dart

```diff
@@ -57,23 +56,45 @@ class _MyAppState extends State<InitialScreen> {
       if (isPro) {
         return CatsScreen();
       } else {
-        return UpsellScreen(
-          offerings: _offerings,
-        );
+        return UpsellScreen();
       }
     }
   }
 }
 
-class UpsellScreen extends StatelessWidget {
-  final Offerings offerings;
+class UpsellScreen extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _UpsellScreenState();
+}
+
+class _UpsellScreenState extends State<UpsellScreen> {
+  Offerings _offerings;
+
+  @override
+  void initState() {
+    super.initState();
+    fetchData();
+  }
+
+  Future<void> fetchData() async {
+    Offerings offerings;
+    try {
+      offerings = await Purchases.getOfferings();
+    } on PlatformException catch (e) {
+      print(e);
+    }
+
```

Comment:
does this need the sim to be logged into to app store / play store?

Reply:
We had it in the initial screen and I moved it to the UpsellScreen. I am not sure about Flutter lifecycle's, but I don't want people to think they have to fetch offerings as soon as the app starts. 

This change is independent from the integration test tho. I was doing UI tests originally, so I actually needed this to not crash the app in the sim. I thought it would be nice to keep it. I can revert it if you think we don't need to do this change now :)